File: FleetMaintenance.Domain/Entities/MaintenanceRequest.cs
using FleetMaintenance.Domain.Enums;

namespace FleetMaintenance.Domain.Entities;

public class MaintenanceRequest
{
    public int Id { get; set; }

    public int VehicleId { get; set; }

    public int MaintenanceTypeId { get; set; }

    public string RequestedByUserId { get; set; } = string.Empty;

    public string RequestedByFullName { get; set; } = string.Empty;

    public string RequestedByEmail { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public DateTime? PreferredDate { get; set; }

    public MaintenanceRequestStatus Status { get; set; } =
        MaintenanceRequestStatus.Pending;

    public DateTime RequestedAt { get; set; } =
        DateTime.UtcNow;

    public DateTime? ReviewedAt { get; set; }

    public string? ReviewedByUserId { get; set; }

    public string? RejectionReason { get; set; }

    public int? MaintenanceRecordId { get; set; }

    public Vehicle Vehicle { get; set; } = null!;

    public MaintenanceType MaintenanceType { get; set; } = null!;

    public MaintenanceRecord? MaintenanceRecord { get; set; }
}

File: fleet-maintenance-client/src/types/maintenanceRequest.ts
export type MaintenanceRequestStatus =
  | "Pending"
  | "Approved"
  | "Rejected"
  | "Cancelled";

export interface MaintenanceRequest {
  id: number;

  vehicleId: number;
  vehiclePlateNumber: string;

  maintenanceTypeId: number;
  maintenanceTypeName: string;

  requestedByUserId: string;
  requestedByFullName: string;
  requestedByEmail: string;

  description: string;
  preferredDate: string | null;

  status: MaintenanceRequestStatus;

  requestedAt: string;
  reviewedAt: string | null;
  reviewedByUserId: string | null;

  rejectionReason: string | null;
  maintenanceRecordId: number | null;
}

export interface MaintenanceRequestFilters {
  search: string;
  status:
    | MaintenanceRequestStatus
    | "";
  fromDate: string;
  toDate: string;
  pageNumber: number;
  pageSize: number;
}

export interface CreateMaintenanceRequestRequest {
  vehicleId: number;
  maintenanceTypeId: number;
  description: string;
  preferredDate?: string | null;
}

export interface ApproveMaintenanceRequestRequest {
  scheduledDate: string;
  dueMileage?: number | null;
  notes?: string | null;
}

export interface RejectMaintenanceRequestRequest {
  reason: string;
}